response_writer.go

```diff
@@ -112,3 +113,12 @@ func (rw *responseWriter) Flush() {
 		flusher.Flush()
 	}
 }
+
+func (rw *responseWriter) Push(target string, opts *http.PushOptions) error {
+	pusher, ok := rw.ResponseWriter.(http.Pusher)
+	if !ok {
+		return fmt.Errorf("the ResponseWriter doesn't support the Pusher interface")
```

Comment:
The use of `fmt.Errorf` seems not useful, would you mind change this line and line 95 to both use `errors.New` instead?

Reply:
Done.